### PR TITLE
HPCC-24626 Add "NoCommon" to STD.File.SprayFixed and Delimited libraries

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add 'noCommon' parameter to Fileservices:Spray*() functions
Implement 'noCommon' parameter passiing to DFUServer.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested manually with Fileservices function call (ECL), ECLWatch and DFUPlus.

### CODE REVIEW

File: dali/ft/filecopy.cpp [hunk 1]
```diff
@@ -3086,6 +3086,7 @@ void FileSprayer::spray()
     }
 
     LOG(MCdebugInfo, job, "compressedInput:%d, compressOutput:%d", compressedInput, compressOutput);
+    LOG(MCdebugInfo, job, "noCommon:%s", (options->getPropBool(ANnocommon) == true ? "true" : "false"));
```
Comment (jakesmith) [inline dali/ft/filecopy.cpp:3089]: minor: use boolToStr:
```
LOG(MCdebugInfo, job, "noCommon:%s", boolToStr(options->getPropBool(ANnocommon)));
```

File: esp/services/ws_fs/ws_fsService.cpp [hunk 2]
```diff
@@ -2088,6 +2088,8 @@ bool CFileSprayEx::onSprayFixed(IEspContext &context, IEspSprayFixed &req, IEspS
         if (!req.getExpireDays_isNull())
             options->setExpireDays(req.getExpireDays());
 
+        options->setNoCommon(req.getNoCommon());
```
Comment (jakesmith) [inline esp/services/ws_fs/ws_fsService.cpp:2091]: This line shouldn't be needed.
There is already code to set it on line 2080.

File: esp/services/ws_fs/ws_fsService.cpp [hunk 3]
```diff
@@ -2270,6 +2272,8 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
...
```
Comment (jakesmith) [inline esp/services/ws_fs/ws_fsService.cpp:2275]: again, there is already code to set, see line 2264..
why was this added?

**Review by jakesmith — COMMENTED** [review]
@AttilaVamos - please see comments

File: dali/ft/filecopy.cpp (hunk 1 above)
Comment (AttilaVamos) [reply dali/ft/filecopy.cpp]: Changed

File: esp/services/ws_fs/ws_fsService.cpp (hunk 2 above)
Comment (AttilaVamos) [reply esp/services/ws_fs/ws_fsService.cpp]: Removed

File: esp/services/ws_fs/ws_fsService.cpp (hunk 3 above)
Comment (AttilaVamos) [reply esp/services/ws_fs/ws_fsService.cpp]: Removed

**Review by jakesmith — COMMENTED** [review]
@AttilaVamos - looks good, but 1 typo in commit and PR message:

"Implemet" -> "Implement"

**Comment by jakesmith** [comment]
@AttilaVamos - the PR message still has the typo'

**Comment by AttilaVamos** [comment]
@jakesmith sorry, fixed.

**Review by jakesmith — APPROVED** [review]
Looks good

**Comment by AttilaVamos** [comment]
@richardkchapman please merge.

**Review by richardkchapman — APPROVED** [review]
